pandas/tslib.pyx

```diff
@@ -1035,7 +1035,11 @@ cdef class _Timestamp(datetime):
 
             # we may be passed reverse ops
             if get_timezone(getattr(self,'tzinfo',None)) != get_timezone(other.tz):
-                    raise TypeError("Timestamp subtraction must have the same timezones or no timezones")
+            
+                if self.tzinfo!="UTC":
+                    self.tz_convert("UTC")
+                if other.tzinfo!="UTC":
+                    other.tz_convert("UTC")
```

Comment:
I don't think it is needed to convert both to UTC, as the underlying value (the ones that get subtracted), already are stored as UTC. Can you check that?